recipe/meta.yaml

```diff
@@ -19,13 +17,19 @@ requirements:
     - python >=2.7
 
 test:
+  requires:
+    - backports.zoneinfo
+    - python 3.8
+    - setuptools 71.0.4
   imports:
     - backports
+    - backports.zoneinfo
+    - setuptools
```

Comment:
This test should be added in `setuptools` and then used as a `downstreams` test.

Reply:
I don't follow. The bug  is in _this_ feedstock, so should be tested here (IMO). Setuptools doesn't require `backports.zoneinfo`, that's just a functional dependent of `backports` that exposes the bug. Happy to be convinced otherwise.

Reply:
setuptools manipulates `sys.path` and `sys.modules` and that's the only reason this test fails. The fix here is okay, but the test should be in setuptools because of its hacky way of vendoring stuff.

Reply:
At the risk of getting into the weeds here (rather than in a setuptools-feedstock issue/PR) the only way to expose this problem is with Python 3.8 (because _this_ feedstock is only really used for that version). However, it would be bad for the setuptools feedstock tests to be pinned to Python 3.8 - is there a way to configure multiple test cases in a conda build?

Reply:
I also agree that it appears as if Setuptools' is implicated here, but I think it's possible to isolate the issue without Setuptools.

Does conda allow you to install packages to a separate directory? Or can you use pip in the test setup? Here's how to create the environment:

```
$ conda create -y -q -n test backports
...
$ conda activate test
$ python3 -m "pip install --target alt-site backports.tarfile"  # or any other backports.* package
$ export PYTHONPATH=$DEFAULT_SITE:alt-site
$ python3 -c "import backports.tarfile"
```

The reason you need $DEFAULT_SITE in the PYTHONPATH is to make sure it appears ahead of `alt-site`.

Presumably that test fails currently and would pass after this fix.